### ISSUE TITLE
Don't show success toast message when failing to delete records

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#2233]
## Steps to reproduce

1. Go to the Table Page for a table that another table references. (E.g. "Publications" is referenced by "Items").

1. Use the row header context menu to delete a record that another record references. The server will respond with an error when making the delete request.

1. Expect no toast message.

1. Observe a success toast message.

![image](https://user-images.githubusercontent.com/42411/211829542-d981ea61-166e-4ef6-8452-8f1dc1369f4e.png)

## Implementation notes

- This will require some refactoring within `RecordsData.deleteSelected` to make callers of that function aware of the failure scenario. Note that the function is designed to delete multiple rows with separate requests, so it's possible that some will fail and some will succeed.

**Comment by pavish** [comment]
I'm adding a backend tag additionally because it is a long standing request to have an endpoint that can delete multiple rows together in a single transaction, and I think we should use that here.

**Comment by kgodey** [comment]
@pavish Please avoid adding additional scope to this issue – if a new backend endpoint is needed, please create a separate issue for that and mark this one as blocked on that.

**Comment by seancolsen** [comment]
+1 on a separate ticket. The two issues can be handled separately I think. I don't even think the backend work needs to block the front end work. I think we should only display the success toast message if _all_ the deletion requests succeed. The original issue in this ticket is a fairly simple mismatch between feedback in the UI after a user action.

**Comment by pavish** [comment]
>  if a new backend endpoint is needed, please create a separate issue for that and mark this one as blocked on that.

Created https://github.com/centerofci/mathesar/issues/2237

> I don't even think the backend work needs to block the front end work. I think we should only display the success toast message if all the deletion requests succeed. The original issue in this ticket is a fairly simple mismatch between feedback in the UI after a user action.

This would be hard to do. We would have to remove the successfully deleted rows from the UI while displaying an error toast if one request fails. Showing a proper feedback will require us to provide individual statuses in some way for each record. A backend API simplifies the UX and UI work, and I think we shouldn't go ahead with the frontend work until we have the API.

**Comment by pavish** [comment]
I misunderstood the issue. This seems to be targeting a single scenario.

I was focused on the scenario in the implementation notes: "Note that the function is designed to delete multiple rows with separate requests, so it's possible that some will fail and some will succeed.".

This issue isn't blocked by the backend work. That is an entirely separate issue of it's own.

**Comment by rajatvijay** [comment]
Reviewing  #2386 reminded me of this issue. @kgodey shouldn't this be a launch blocker? If not we should at least remove the toast even for the success condition since it gives the wrong messaging to the user.

**Comment by kgodey** [comment]
@rajatvijay I'm reserving launch blocker status for stuff that will prevent our target users from adopting Mathesar. This is too minor of an issue for that. I've set the priority to "high" though.

**Comment by kgodey** [comment]
@rajatvijay are you actively working on this? if not, consider assigning to @DanyaalMahmood if you think it would be suitable for a contributor.

**Comment by rajatvijay** [comment]
Sure @kgodey done. @DanyaalMahmood I have assigned this to you now. You can start working on it and let me know if you have any questions.

**Comment by DanyaalMahmood** [comment]
@seancolsen @kgodey @rajatvijay. I have opened a PR, can you guys please review it?
